apps/recorder/widget.js

```diff
@@ -239,7 +247,7 @@
     reload();
     Bangle.drawWidgets(); // relayout all widgets
   },isRecording:function() {
-    return !!writeInterval;
+    return isRecording;
```

Comment:
I'm a bit unsure about my change to introducing and using the variable `isRecording`.

Reply:
I suppose we could repurpose `writeInterval`, so it's either a number (the interval) or `true` in which case we know we're using gps as the driver for the writes. We could call it `writeSetup` and we'd just need to make sure we handle the three cases in each place it's used (`number | true | undefined`)

Possibly controversial, but typescript would help if we wanted to really go whole hog

Reply:
I did the change to `writeSetup`. Thanks :+1: 

As for typescript, I don't know it so don't have anything to say about it :P

Reply:
That's fair! :D